stdlib/mexpr/boot-parser.mc

```diff
@@ -138,6 +138,45 @@ lang BootParser = MExprAst
      TmNever {ty = TyUnknown(),
               info = ginfo t 0}
 
+  -- Get type help function
+  sem gtype(t:Unknown) =
+  | n -> let t2 = bootParserGetType t n in
+        matchType t2 (bootParserGetId t2)
+
+  sem matchType (t:Unknown) =
+  | 200 /-TyUnknown-/ ->
+    TyUnknown {info = ginfo t 0}
+  | 201 /-TyBool-/ ->
+    TyBool {info = ginfo t 0}
+  | 202 /-TyInt-/ ->
+    TyInt {info = ginfo t 0}
+  | 203 /-TyFloat-/ ->
+    TyFloat {info = ginfo t 0}
+  | 204 /-TyChar-/ ->
+    TyChar {info = ginfo t 0}
+  | 205 /-TyArrow-/ ->
+    TyArrow {info = ginfo t 0,
+             from = gtype t 0,
+             to = gtype t 1}
+  | 206 /-TySeq-/ ->
+    TySeq {info = ginfo t 0,
+           ty = gtype t 0}
+  | 207 /-TyRecord-/ ->
+    let lst = makeSeq (lam n. (gstr t n, gtype t n)) (glistlen t 0) in
+    TyRecord {info = ginfo t 0,
+              fields = mapFromList cmpSID (map (lam b. (stringToSid b.0, b.1)) lst)}
+  | 208 /-TyVariant-/ ->
+    let lst = makeSeq (lam n. (gname t n, gtype t n)) (glistlen t 0) in
```

Comment:
In `boot` you are only passing a list of strings (since there are no types in variants there, that seems right), but here you are reading both a string and a type so this would not work for non-empty variants. Perhaps it would be better to have the boot-parser report an error if the `TyVariant` is non-empty, as we do in `pprint.mc`? I'm not sure what we want to do here, so let's discuss this at the meeting.